ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_creation.py

```diff
@@ -306,3 +308,56 @@ def test_eye_like(
         device=on_device,
         ground_truth_backend=ground_truth_backend,
     )
+
+
+@st.composite
+def _get_dtype_buffer_count_offset(draw):
+    dtype, value = draw(
+        helpers.dtype_and_values(
+            available_dtypes=helpers.get_dtypes("valid"),
+        )
+    )
+    length = value[0].size
+    value = value[0].tobytes()
+
+    offset = draw(helpers.ints(min_value=0, max_value=length - 1))
+    count = draw(helpers.ints(min_value=-2 ** 30, max_value=length - offset))
+    if count == 0:
+        count = -1
+    offset = offset * np.dtype(dtype[0]).itemsize
+
+    return dtype, value, count, offset
+@handle_test(
+    fn_tree="functional.ivy.experimental.frombuffer",
+    dtype_buffer_count_offset=_get_dtype_buffer_count_offset(),
+    test_instance_method=st.just(False),
+    test_with_out=st.just(False),
+    test_gradients=st.just(False),
+)
+
+
+def test_frombuffer(
+    dtype_buffer_count_offset,
+    test_flags,
+    backend_fw,
+    fn_name,
+    on_device,
+    ground_truth_backend,
+):
+    input_dtype, buffer, count, offset = dtype_buffer_count_offset
+    dtype = input_dtype
+    if isinstance(dtype, list):
+        dtype = ivy.Dtype(dtype[0])
```

Comment:
are you sure this is necessary? I think `input_dtype` can be directly used

Reply:
Hi @MahmoudAshraf97 . The input_dtype returns a list of dtype while backend frombuffer expects dtype. So, I just wanna take the first element.